src/devices/vdml_motors.cpp

```diff
@@ -13,9 +13,72 @@
 #include "kapi.h"
 #include "pros/motors.hpp"
 
+#include <vector>
+
+/**
+ * Macro to claim the motor group mutex with the error code being PROS_ERR
+ * 
+ */
+#define claim_mg_mutex(error)						 			 \
+	if (!_motor_group_mutex.take(TIMEOUT_MAX)) { \
+		return error;															 \
+	}																						 
+
+/**
+ * Macro to free the motor group mutex with the error value being PROS_ERR
+ * 
+ */
+#define give_mg_mutex(error)									 \
+	if (!_motor_group_mutex.give()) {            \
+		return error;                         	   \
+	}
+
+/**
+ * Macro to loop through a function call for each motor in a motor group
+ * with an error value of PROS_ERR
+ * 
+ */
+#define mg_foreach(func_call, motors)                       	\
```

Comment:
instead of having "out" as something we expect, can we have this as a macro parameter?